src/NodaTime.Test/TimeZones/BclDateTimeZoneTest.cs

```diff
@@ -37,15 +40,34 @@ public class BclDateTimeZoneTest
             TimeZoneInfo.CreateCustomTimeZone(zone.Id, zone.BaseUtcOffset, zone.DisplayName, zone.StandardName,
                 zone.DisplayName, zone.GetAdjustmentRules());
 
+        // TODO: Check what this does on Mono, both on Windows and Unix.
+
+        [Test]
+        [TestCaseSource(nameof(BclZonesAndIds))]
+        public void AreWindowsStyleRules(TimeZoneInfo zone, string id)
+        {
+            var expected = !TestHelper.IsRunningOnDotNetCoreUnix;
+            var rules = zone.GetAdjustmentRules();
+            if (rules == null || rules.Length == 0)
+            {
+                return;
+            }
+            Assert.AreEqual(expected, BclDateTimeZone.AreWindowsStyleRules(rules));
+        }
+
         [Test]
-        [TestCaseSource(nameof(BclZones))]
+        [TestCaseSource(nameof(BclZonesAndIds))]
         [Category("BrokenOnMonoLinux")]
-        public void AllZoneTransitions(TimeZoneInfo windowsZone)
+        public void AllZoneTransitions(TimeZoneInfo windowsZone, string id)
         {
             var nodaZone = BclDateTimeZone.FromTimeZoneInfo(windowsZone);
 
+            // Currently .NET Core doesn't expose the information we need to determine any DST recurrence
+            // after the final tzif rule. (Bug to be filed against corefx.) For the moment, limit how far we check.
```

Comment:
(is this referring to one of the bugs you already filed?)

Reply:
Will update to link to the existing one.